internal/variables.go

```diff
@@ -69,6 +69,25 @@ type K3aiRootPlugin struct {
 
 	Resources []string `yaml:"resources,omitempty"`
 }
+
+type K3aiConfig struct {
+	Api     string `yaml:"api"`
+	Kind    string `yaml:"kind"`
+	Cluster struct {
+		Name string `yaml:"name"`
+		Type string `yaml:"type"`
+	}
+	Plugin struct {
+		Name   string `yaml:"name"`
+		Target string `yaml:"target"`
+	}
+	Run struct {
+		Source  string `yaml:"source"`
+		Target  string `yaml:"target"`
+		Backend string `yaml:"backend"`
+	}
+}
+
```

Comment:
The current definition doesn't deal with artifact path.

This is due to the reason that artifact collection adds a whole new level of complexity to this PR, and hence, it could be handled in a new PR, or in a subsequent release.

Reply:
I agree, I think we have multiple challenges there:
1) Backend used means retrieve artifacts in different ways (I assume MLFlow and KFP both use minio as default but could be different in the future)
2) Where we do store artifacts? I would assume we may grab them as they are and save them in a folder in GH but, what if is too big the artifact?
3) Artifact are generated automatically but metrics not (i.e.: MLFlow ) so we probably can't save everything

Reply:
1. MLFlow supports [4 ways to store artifacts](https://mlflow.org/docs/latest/tracking.html#how-runs-and-artifacts-are-recorded) including SQLite and local storage. And yes, retrieving artifacts for different plugins is tricky.
2. We can store artifacts on:
    - different cloud providers through configs
    - on a separate Github repo for metrics (PAT needs to be passed through Github Actions)
3. Yes, saving metrics is something to work upon.

Reply:
Github Actions supports uploading artifacts through their [own custom action](https://github.com/actions/upload-artifact).